### PR TITLE
Merge 18.4.0.2 into `develop`

### PR DESCRIPTION
**Pull request by mokagio**
Includes:

- Gutenberg 1.63.1 ([#17302](https://github.com/wordpress-mobile/WordPress-iOS/pull/17302)) by @dcalhoun 

Conflicts on `Podfile` and `Podfile.lock`, I solved them by keeping 1.63.1, the version on the release branch. On `develop`, we a 1.64.0 pre-release but it doesn't yet contain those fixes. They'll come in when a new pre-release or a stable version of 1.64.0 is cut, as they are already in Gutenberg mobile's develop: wordpress-mobile/gutenberg-mobile#4113.

Nothing to test, if CI Shows green, we're good.

## Regression Notes
1. Potential unintended areas of impact
N.A.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N.A.

3. What automated tests I added (or what prevented me from doing so)
N.A.

---

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes. **N.A.**
- [x] I have considered adding accessibility improvements for my changes. **N.A.**
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary. **N.A.**

### CODE REVIEW

**Comment by AliSoftware** [comment]
> Conflicts on Podfile and Podfile.lock, I solved them by keeping 1.63.1, the version on the release branch. On develop, we a 1.64.0 pre-release but it doesn't yet contain those fixes. 

@mokagio I usually solve it the opposite way, keeping what's in `develop` to resolve the conflict. Because if you make `1.63.1` land in `develop`, any new feature worked on in `develop` which depends on what was introduced in `1.64.0-alpha*` (a very likely scenario to exist, otherwise why would they have landed the alpha in develop already if no new feature needed it?) will break if you make develop revert to `1.63.1`.

The team usually integrate a new version in `develop` very soon after fixing things in `release` — so that develop ultimately include the same fix — so imho better not break `develop`, and then wait for the fix to be duplicated there soon, rather than introducing a regression in `develop` via this PR landing.

Would love to hear @dcalhoun confirm (or deny) this though.

File: Podfile [hunk 1]
```diff
@@ -166,7 +166,7 @@ abstract_target 'Apps' do
     ## Gutenberg (React Native)
     ## =====================
     ##
-    gutenberg :tag => 'v1.64.0-alpha2'
+    gutenberg :tag => 'v1.63.1'
```
Comment (AliSoftware) [inline Podfile:169]: This feels like the wrong conflict resolution to me (see previous comment)

**Review by AliSoftware — CHANGES_REQUESTED** [review]

File: Podfile (hunk 1 above)
Comment (dcalhoun) [reply Podfile:169]: Agreed, the correct conflict resolution would be the newer tag within `develop` — `1.64.0-alpha2`.

**Comment by dcalhoun** [comment]
> Because if you make `1.63.1` land in `develop`, any new feature worked on in `develop` which depends on what was introduced in `1.64.0-alpha*` (a very likely scenario to exist, otherwise why would they have landed the alpha in develop already if no new feature needed it?) will break if you make develop revert to `1.63.1`.

This is correct. The very existence of an alpha tag in the host WordPress app repos means _something_ in Gutenberg is dependent upon the changes in the alpha release. Reverting from an alpha to a lower release will likely break the Gutenberg `develop` branch for local development against the WordPress host apps. 

> The team usually integrate a new version in `develop` very soon after fixing things in `release` — so that develop ultimately include the same fix — so imho better not break `develop`, and then wait for the fix to be duplicated there soon, rather than introducing a regression in `develop` via this PR landing.

This is somewhat true. For better or worse, we only land alpha releases into the WordPress host apps when Gutenberg _depends_ upon the changes the release contains. For the fix in `1.63.1`, it was a fix for something completely contained in Gutenberg. So, it is unlikely we will land it into the WordPress apps `develop` until (A) the next release (`1.64.0`) or (B) an alpha release for an unrelated feature (`1.65.0-alpha.X`). 

If this practices causes confusion or trouble, we could discuss immediately landing _any_ change, but we avoid doing so because it is currently a fair amount of manual work to land changes in both WordPress host apps. 😩

**Comment by mokagio** [comment]
Lol and that's why I shouldn't open merge PR in the evening 🤦‍♂️

**Pull request closed without merging**